bee-common/bee-packable/src/packable/vec_prefix.rs

```diff
@@ -8,117 +8,81 @@ use crate::{
     error::{PackError, PackPrefixError, UnpackError, UnpackPrefixError},
     packer::Packer,
     unpacker::Unpacker,
-    Packable,
+    Bounded, BoundedU16, BoundedU32, BoundedU64, BoundedU8, InvalidBoundedU16, InvalidBoundedU32, InvalidBoundedU64,
+    InvalidBoundedU8, Packable,
 };
 
 use alloc::vec::Vec;
 use core::{convert::TryFrom, marker::PhantomData};
 
-/// Error encountered when attempting to convert a [`Vec<T>`] into a [`VecPrefix`], where
-/// the length of the source vector exceeds the maximum length of the [`VecPrefix`].
-#[derive(Debug, PartialEq, Eq)]
-pub struct PrefixedFromVecError {
-    /// Maximum length of the [`VecPrefix`].
-    pub max_len: usize,
-    /// Actual length of the source vector.
-    pub actual_len: usize,
-}
-
 /// Wrapper type for [`Vec<T>`] where the length prefix is of type `P`.
 /// The [`Vec<T>`]'s maximum length is provided by `N`.
 #[derive(Clone, Debug, Eq, PartialEq)]
 #[repr(transparent)]
-pub struct VecPrefix<T, P, const N: usize> {
+pub struct VecPrefix<T, B: Bounded> {
     inner: Vec<T>,
-    marker: PhantomData<P>,
+    bounded: PhantomData<B>,
 }
 
-impl<T, P, const N: usize> VecPrefix<T, P, N> {
-    /// Creates a new empty [`VecPrefix<T, P>`].
-    pub fn new() -> Self {
-        Self {
-            inner: Vec::new(),
-            marker: PhantomData,
+macro_rules! impl_vec_prefix {
+    ($ty:ident, $bounded:ident, $err:ident) => {
+        impl<T, const MIN: $ty, const MAX: $ty> VecPrefix<T, $bounded<MIN, MAX>> {
+            /// Creates a new empty [`VecPrefix<T, P>`] with a specified capacity.
+            pub fn with_capacity(capacity: usize) -> Self {
```

Comment:
Your PR description says it's been removed ?

Reply:
🤔